samples/media/video/src/main/java/com/example/platform/media/video/StyleTransferShaderProgram.java

```diff
@@ -0,0 +1,229 @@
+            String predictModel = "predict_float16.tflite";
+            String transferModel = "transfer_float16.tflite";
+            Interpreter predictInterpeter =
+                    new Interpreter(FileUtil.loadMappedFile(context, predictModel), options);
+            transformInterpreter =
+                    InterpreterApi.create(FileUtil.loadMappedFile(context, transferModel), options);
+            int inputPredictTargetHeight = predictInterpeter.getInputTensor(0).shape()[1];
+            int inputPredictTargetWidth = predictInterpeter.getInputTensor(0).shape()[2];
+            int[] outputPredictShape = predictInterpeter.getOutputTensor(0).shape();
+
+            inputTransformTargetHeight = transformInterpreter.getInputTensor(0).shape()[1];
+            inputTransformTargetWidth = transformInterpreter.getInputTensor(0).shape()[2];
+            outputTransformShape = transformInterpreter.getOutputTensor(0).shape();
+
+            InputStream inputStream = context.getAssets().open(styleAssetFileName);
+            Bitmap styleImage = BitmapFactory.decodeStream(inputStream);
+            inputStream.close();
+            TensorImage styleTensorImage =
+                    getScaledTensorImage(styleImage, inputPredictTargetWidth, inputPredictTargetHeight);
+            predictOutput = TensorBuffer.createFixedSize(outputPredictShape, DataType.FLOAT32);
+            predictInterpeter.run(styleTensorImage.getBuffer(), predictOutput.getBuffer());
+        } catch (IOException | GlUtil.GlException e) {
+            Log.w("DEBUG", "Error setting up TfShaderProgram", e);
+            throw new VideoFrameProcessingException(e);
+        }
+    }
+
+    @Override
+    public Size configure(int inputWidth, int inputHeight) {
+        width = inputWidth;
+        height = inputHeight;
+        return new Size(inputWidth, inputHeight);
+    }
+
+    @Override
+    public void drawFrame(int inputTexId, long presentationTimeUs)
+            throws VideoFrameProcessingException {
+        ByteBuffer pixelBuffer = ByteBuffer.allocateDirect(width * height * 4);
+
+        Bitmap bitmap;
+        int texId;
+        try {
+            int[] boundFramebuffer = new int[1];
+            GLES20.glGetIntegerv(GLES20.GL_FRAMEBUFFER_BINDING, boundFramebuffer, /* offset= */ 0);
+
+            int fboId = GlUtil.createFboForTexture(inputTexId);
+            GlUtil.focusFramebufferUsingCurrentContext(fboId, width, height);
+            GLES20.glReadPixels(
+                    /* x= */ 0,
+                    /* y= */ 0,
+                    width,
+                    height,
+                    GLES20.GL_RGBA,
+                    GLES20.GL_UNSIGNED_BYTE,
+                    pixelBuffer);
+            GlUtil.checkGlError();
+            bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
+            bitmap.copyPixelsFromBuffer(pixelBuffer);
+
+            Log.w("DEBUG", "Process frame at " + (presentationTimeUs / 1000) + " ms");
```

Comment:
Should we follow the same convention as in other classes and define a TAG const and use it instead of "Debug"?

Reply:
that makes sense, will do